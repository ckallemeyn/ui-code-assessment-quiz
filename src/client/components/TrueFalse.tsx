import React, {  useContext } from 'react'
import { FormField, Radio } from 'semantic-ui-react'
import QuestionContext from '../contexts/QuestionContext'


const TrueFalse = () => {
  const { currentAnswer, handleChange }: any = useContext(QuestionContext)

  return (
    <>
      <FormField>
        <Radio
          label="True"
          name="radioGroup"
          value="True"
          checked={currentAnswer === "True"}
          onChange={handleChange}
        />
      </FormField>
      <FormField>
        <Radio
          label="False"
          name="radioGroup"
          value="False"
          checked={currentAnswer === "False"}
          onChange={handleChange}
        />
      </FormField>
    </>
  );
}

export default TrueFalse
