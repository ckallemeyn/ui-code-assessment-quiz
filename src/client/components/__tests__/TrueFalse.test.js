import React from 'react'
import { render } from "@testing-library/react";
import TrueFalse from "../TrueFalse";

describe("TrueFalse Answer Option", () => {
  test("it should render", () => {
    const { container } = render(<TrueFalse />);

    expect(container).toMatchSnapshot();
  });
});
