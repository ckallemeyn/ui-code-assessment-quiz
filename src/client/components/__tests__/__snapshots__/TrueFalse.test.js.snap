// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrueFalse Answer Option it should render 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="ui radio checkbox"
      data-testid="radio-btn"
    >
      <input
        class="hidden"
        name="radioGroup"
        readonly=""
        tabindex="0"
        type="radio"
        value="True"
      />
      <label>
        True
      </label>
    </div>
  </div>
  <div
    class="field"
  >
    <div
      class="ui radio checkbox"
    >
      <input
        class="hidden"
        name="radioGroup"
        readonly=""
        tabindex="0"
        type="radio"
        value="False"
      />
      <label>
        False
      </label>
    </div>
  </div>
</div>
`;
